File: #1508.range-sum-of-sorted-subarray-sums.cpp
class Solution {
public:
    int rangeSum(vector<int>& nums, int n, int left, int right) {

        priority_queue<pair<int, int>, vector<pair<int, int>>, greater<pair<int, int>>> pq;
        for(int i = 0; i < n; ++i){
            pq.push({nums[i], i + 1});
        }

        int result = 0;
        int modulo = 1e9 + 7;
        for (int i = 1; i <= right; ++i) {
            pair<int, int> p = pq.top(); pq.pop();
            
            if (i >= left) {
                result = (result + p.first) % modulo;
            }
            
            if (p.second < n) {
                pq.push({p.first + nums[p.second++], p.second});
            }
        }
        
        return result;
    }
};

File: #1508.range-sum-of-sorted-subarray-sums.js
/**
 * @param {number[]} nums
 * @param {number} n
 * @param {number} left
 * @param {number} right
 * @return {number}
 */
var rangeSum = function(nums, n, left, right) {
    const minqueue = new MinPriorityQueue({priority : x=> x[0]});
    for (let i = 0; i < n; i++) {
        minqueue.enqueue([nums[i], i]);
    }

    let mod = Math.pow(10, 9) + 7;
    let result = 0;
    for (let i = 1; i <= right; i++) {
        let [sum,index] = minqueue.dequeue().element;
        if (i >= left) {
            result = (result + sum) % mod;
        }
        if (index < n - 1) {
            minqueue.enqueue([sum + nums[index + 1],index + 1]);
        }
    }
    return result;
};
